### PR TITLE
Update Chat-O-Matic, libpurple, purple_discord

### PR DESCRIPTION
**Pull request by Peppersawce**
I was trying Chat-O-Matic on x64 (the package from HaikuDepot) and it could do nothing but crash and burn, also it lacked an icon.

I think it's a shame that one of the few native messaging apps, and one that through libpurple could actually support a lot of different protocols, is downloadable broken from the repositories so I tweaked the recipe a bit, pulling from the last git commit. It builds, runs and connects fine, at least to IRC, libpurple and extensions show up but are mostly untested.

Untested on x86 but I suppose it builds and works fine? Let me know

### CODE REVIEW

File: haiku-apps/chat_o_matic/chat_o_matic-0.0.2~git.recipe [hunk 1]
```diff
@@ -70,12 +71,6 @@ INSTALL()
 
 	addAppDeskbarSymlink $appsDir/Chat-O-Matic
 
-	mkdir -p $includeDir/chat-o-matic/
-	cp application/ChatOMatic.h application/ChatProtocol.h \
```
Comment (mtl1979) [inline haiku-apps/chat_o_matic/chat_o_matic-0.0.2~git.recipe]: `application/ChatOMatic.h` was removed

Comment (Peppersawce) [reply haiku-apps/chat_o_matic/chat_o_matic-0.0.2~git.recipe]: Yep, that's why I removed it as well :)

Comment (Peppersawce) [reply haiku-apps/chat_o_matic/chat_o_matic-0.0.2~git.recipe]: Fixed, ty

File: dev-libs/libpurple/libpurple-2.14.13.recipe [hunk 2]
```diff
@@ -3,20 +3,22 @@ DESCRIPTION="libpurple is a library for IM communication used by many open \
...
 ARCHITECTURES="all !x86_gcc2"
 SECONDARY_ARCHITECTURES="x86"
 
-libVersion=0.14.10
+libVersionVar="`echo "$portVersion" | cut -b3-7`"
+libVersion=0.${libVersionVar}
 libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
 
-pythonVersion=3.9
+pythonVersion=3.12 # 3.9 works as well
```
[the remark this answers is not in the document]
Comment (Peppersawce) [reply dev-libs/libpurple/libpurple-2.14.13.recipe]: Alright, thank you for the help :)

**Pull request merged**